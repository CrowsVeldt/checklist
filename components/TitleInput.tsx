import { StyleSheet, Text, TextInput, View } from "react-native";

export default function TitleInput({
  title,
  setTitle,
}: {
  title: string;
  setTitle: any;
}) {
  return (
    <View style={styles.titleInputWrapper}>
      <Text style={styles.titleLabel}>Title:</Text>
      <TextInput
        style={styles.titleInput}
        value={title}
        maxLength={16}
        onChangeText={setTitle}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  titleInputWrapper: {
    flexDirection: "row",
  },
  titleInput: {
    borderWidth: 1,
    borderColor: "black",
    borderRadius: 10,
    width: "60%",
  },
  titleLabel: {
    width: "20%",
    textAlign: "center",
    textAlignVertical: "center",
    padding: 10,
  },
});
