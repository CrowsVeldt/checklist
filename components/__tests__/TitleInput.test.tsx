import { render } from "@testing-library/react-native";
import TitleInput from "../TitleInput";

describe("<TitleInput />", () => {
  test("TitleInput renders correctly", () => {
    const tree = render(
      <TitleInput title="title" setTitle={() => {}} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
