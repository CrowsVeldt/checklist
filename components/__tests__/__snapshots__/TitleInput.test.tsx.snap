// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleInput /> TitleInput renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      {
        "padding": 10,
        "textAlign": "center",
        "textAlignVertical": "center",
        "width": "20%",
      }
    }
  >
    Title:
  </Text>
  <TextInput
    maxLength={16}
    onChangeText={[Function]}
    style={
      {
        "borderColor": "black",
        "borderRadius": 10,
        "borderWidth": 1,
        "width": "60%",
      }
    }
    value="title"
  />
</View>
`;
